postgres/tests/test_statements.py

```diff
@@ -414,6 +415,95 @@ def test_statement_samples_collect(
         conn.close()
 
 
+@pytest.mark.parametrize("pg_stat_activity_view", ["pg_stat_activity", "datadog.pg_stat_activity()"])
+@pytest.mark.parametrize(
+    "user,password,dbname,query,arg,expected_out,expected_keys",
+    [
+        (
+            "bob",
+            "bob",
+            "datadog_test",
+            "SELECT city FROM pg_sleep(3), persons WHERE city = %s",
+            "hello",
+            {
+                'datname': 'datadog_test',
+                'usename': 'bob',
+                'state': 'active',
+                'query': "d9193c18a6f372d8",
+            },
+            ["xact_start", "query_start", "pid", "client_port", "client_addr"],
+        ),
+    ],
+)
+def test_activity_samples_collection(
+    aggregator,
+    integration_check,
+    dbm_instance,
+    pg_stat_activity_view,
+    user,
+    password,
+    dbname,
+    query,
+    arg,
+    expected_out,
+    expected_keys,
+):
+    dbm_instance['pg_stat_activity_view'] = pg_stat_activity_view
+    check = integration_check(dbm_instance)
+    check._connect()
+
+    # Run a query in async mode so the test will catch it as its executing.
+    conn = psycopg2.connect(host=HOST, dbname=dbname, user=user, password=password, async_=True)
```

Comment:
I don't love this approach to testing, but this was the best way I could get a good test of entire code path without having to mock everything. In the final version we can just mock what `_get_new_pg_stat_activity()` returns.

Reply:
Could ensure we "catch" the query by opening a transaction, then the state will be "idle in transaction" which we should be capturing (see my other comment). 

That would be much more robust then trying to catch it in "active" state.